selene_sdk/samplers/random_positions_sampler.py

```diff
@@ -151,86 +159,214 @@ def __init__(self,
             save_datasets=save_datasets,
             output_dir=output_dir)
 
-        self._sample_from_mode = {}
-        self._randcache = {}
-        for mode in self.modes:
-            self._sample_from_mode[mode] = None
-            self._randcache[mode] = {"cache_indices": None, "sample_next": 0}
-
         self.sample_from_intervals = []
         self.interval_lengths = []
 
-        if self._holdout_type == "chromosome":
-            self._partition_genome_by_chromosome()
+        self._num_chroms = 0
+        self._genome_n_bases = 0
+        # Holds the size of the parition of each mode (the portion that is
+        # specified to belong to each mode)
+        self._parititon_sizes = {"validate" : 0, "train" : 0, "test" : 0}
+
+        self._validation_holdout = validation_holdout
+        self._N_validation = validation_size
+
+        if test_holdout:
```

Comment:
I think you are doing a lot of extra work in this initialization, a lot of it is done in online sampler https://github.com/FunctionLab/selene/blob/master/selene_sdk/samplers/online_sampler.py#L157 
probably google how superclass or class inheritance works